src/systems/buoyancy/Buoyancy.cc

```diff
@@ -114,6 +226,44 @@ void Buoyancy::Configure(const Entity &_entity,
   {
     this->dataPtr->fluidDensity = _sdf->Get<double>("uniform_fluid_density");
   }
+  else if(_sdf->HasElement("graded_buoyancy"))
+  {
+    this->dataPtr->buoyancyType =
+      BuoyancyPrivate::BuoyancyType::GRADED_BUOYANCY;
+
+    auto gradedElement = _sdf->GetFirstElement();
```

Comment:
If there are other elements, you may get them by mistake. I think this is safer:

```suggestion
    auto gradedElement = _sdf->GetElement("graded_buoyancy");
```

Reply:
`sdf` is `const`. I had gone down this route earlier. Unfortunately `GetElement` is not `const`. It will create an element if it does not exist. I don't like the way its named but :shrug:.

> A new child element, with the provided name, is added to this element if there is no existing child element. 

http://osrf-distributions.s3.amazonaws.com/sdformat/api/11.2.0/classsdf_1_1v11_1_1Element.html#a4cf1ee6d6b0cec87d448e4a5968c4307

Reply:
It should be fine for now as it is the only argument that is taken by the plugin. In future we will have to wrap it in a loop.

Reply:
Ahh true, we recently added `FindElement` to work around that (https://github.com/ignitionrobotics/sdformat/pull/620). But no need to use it here, it was just a small suggestion.